src/main/java/seedu/ddd/model/contact/common/ContactEmailPredicate.java

```diff
@@ -0,0 +1,19 @@
+package seedu.ddd.model.contact.common;
+
+import java.util.function.Predicate;
+
+/**
+ * Tests that a {@Code Contact} 's {@code Email} matches the given phone number partially or fully.
+ */
+public class ContactEmailPredicate implements Predicate<Contact> {
+    private final Email email;
+
+    public ContactEmailPredicate(Email email) {
+        this.email = email;
+    }
+
+    @Override
+    public boolean test(Contact contact) {
+        return this.email.value.contains(contact.getEmail().value);
+    }
```

Comment:
Should we just take a string argument here instead for flexibility? Because there is format validation is the `Email` constructor, which means that the search must match the `VALIDATION_REGEX`.

Reply:
i could change it to this.email.matches(contact.getEmail()) since both are instances of email